compiler/rustc_middle/src/ty/query.rs

```diff
@@ -106,6 +106,55 @@ impl<'tcx> TyCtxt<'tcx> {
     }
 }
 
+#[inline(always)]
+fn query_get_at<'tcx, Cache, K>(
+    tcx: TyCtxt<'tcx>,
+    execute_query: fn(
```

Comment:
The regression probably comes from here: an indirect call (fn pointer) to an indirect call (`dyn QueryEngine`).

Reply:
There's only 1 indirect call, same as before.

Reply:
How?

Reply:
Actually you're right. LLVM will inline it anyway though as `query_get_at` is `#[inline(always)]` so it will discover the function it points to.